core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -241,10 +242,16 @@
     @Setter(AccessLevel.NONE)
     private boolean shouldBeReused = false;
 
+    @Deprecated
     public GenericContainer() {
         this(TestcontainersConfiguration.getInstance().getTinyImage());
     }
 
+    public GenericContainer(DockerImageName dockerImageName) {
+        this.image = new RemoteDockerImage(dockerImageName);
+    }
+
+    @Deprecated
```

Comment:
what about `GenericContainer(Future)`? Since it can be replaced with `new GenericContainer(new RemoteDockerImage(future))`

Reply:
Ooh, interesting suggestion - I'll explore that.

Reply:
I've not done this yet.

Reply:
I've had a quick try, but I'm not sure about this. 

This example from existing code:
```
new GenericContainer(new ImageFromDockerfile() ...
```
would become 
```
new GenericContainer(new RemoteDockerImage(new ImageFromDockerfile() ...
```

which I think is quite ugly - it's really not a 'remote docker image', so naming-wise it's a bit of a stretch.

I wonder if, really, the way to make this work would be to rename `RemoteDockerImage` to `FutureDockerImageName`, which more accurately reflects what it's used for. So the example would become:

```
new GenericContainer(new FutureDockerImageName(new ImageFromDockerfile() ...
```

We'd of course keep `RemoteDockerImage` around, deprecated, as no more than an alias to `FutureDockerImageName`.

Two problems I see with this that I don't think we can magic away. I think I could live with these:

* the three levels of constructors involved are a bit clunky, but it's not a big deal
* it's such as shame that type erasure prevents us from overloading the `GenericContainer(Future<String>)` constructor, because otherwise `FutureDockerImageName` could just be `Future<DockerImageName>` 😂  😭

Reply:
Yeah, I agree that "Remote" is misleading here.

But the example you provided made me think whether we should even wrap `ImageFromDockerfile` with `RemoteDockerImage`, since it is definitely local, does not need an auth resolve, etc etc...
what if we extract a common interface from `RemoteDockerImage` and make `ImageFromDockerfile` implement it?

Reply:
With the interface being `DockerImage`?

Reply:
As discussed on Slack, we'll defer this to a separate PR.

It feels like more extensive and complicated changes are needed to refactor the Remote/Future images code, and I feel it would be sensible to not add that complexity on top of this, already massive, PR.